### PR TITLE
Add Jenkins e2e script

### PR DESCRIPTION
**Pull request by zmerlynn**
This commit adds and documents the e2e script that Jenkins runs

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+tar -xzf kubernetes.tar.gz
+tar -xzf kubernetes-test.tar.gz
+cd kubernetes
+
+# This variable is only relevant on GKE-CI
+export CLUSTER_API_VERSION=$(echo ${GITHASH} | cut -c 2-)
+
+go run ./hack/e2e.go ${E2E_OPT} -v -down
+go run ./hack/e2e.go ${E2E_OPT} -v -up
+go run ./hack/e2e.go -v -ctl="version --match-server-version=false"
```
Comment (lavalamp) [inline hack/jenkins/e2e.sh]: Why is "match-server-version=false" needed here?

Comment (zmerlynn) [inline hack/jenkins/e2e.sh]: The `kubernetes-e2e-gke` job is version skewed, and `hack/e2e.go` adds `--match-server-version` before calling `kubectl`, so this is basically counteracting it.

Comment (lavalamp) [inline hack/jenkins/e2e.sh]: so this is testing a new client on an old cluster?

Comment (zmerlynn) [inline hack/jenkins/e2e.sh]: That's one of the jobs, and it also has to explicitly change E2E_OPT. I'm
just logging the client/server versions here, so really just want something
that won't error out regardless.

**Comment by lavalamp** [comment]
LGTM

**Pull request merged**